src/modules/auth/services/auth.service.ts

```
@@ -0,0 +1,49 @@
+}
+
+@Injectable()
+export class AuthService {
+  private users: User[] = [];
+  private nextId = 1;
+
+  constructor(private readonly hashingService: HashingService) {}
+
+  async register(username: string, password: string): Promise<User> {
+    // Check for duplicate username
+    if (this.users.find(user => user.username === username)) {
```

ESLint

[error] Replace `user` with `(user)`

(prettier/prettier)